libs/api/repository/src/lib/gn4/gn4-repository.ts

```diff
@@ -137,6 +141,27 @@ export class Gn4Repository implements RecordsRepositoryInterface {
       )
   }
 
+  getFeatureCatalog(
+    metadataUuid: string,
+    approvedVersion?: boolean
+  ): Observable<DatasetFeatureCatalog | null> {
+    return this.gn4RecordsApi
+      .getFeatureCatalog(metadataUuid, approvedVersion)
```

Comment:
I honestly did not know this API existed. Looking at the Java code I think it might not do exactly what we need. Instead, if the record is provided as argument, we can simply look for `record.extras.featureCatalogs` and do the mapping on this.